src/ert/storage/local_ensemble.py

```diff
@@ -357,15 +353,24 @@ def load_gen_data_as_df(
         return pd.concat(dfs)
 
     def load_field(self, key: str, realizations: List[int]) -> npt.NDArray[np.double]:
-        result = []
+        result: Optional[npt.NDArray[np.double]] = None
         for realization in realizations:
             input_path = self.mount_point / f"realization-{realization}"
             if not input_path.exists():
                 raise KeyError(f"Unable to load FIELD for key: {key}")
             data = np.load(input_path / f"{key}.npy")
-            data_no_nans = data[~np.isnan(data)]
-            result.append(data_no_nans)
-        return np.stack(result).T  # type: ignore
+            data = data[np.isfinite(data)]
+
+            if result is None:
+                result = np.ndarray((len(realizations), data.size))
```

Comment:
What if we instead initialize with this:

`result = np.empty((data.size, len(realizations)), dtype=np.double)

and then do

`result[:, realization] = data`

to avoid having to transpose before returning.

Transpose is cheap, but this is still more explicit I think.

Reply:
I only did it to keep it the same as the original. The transpose may return a view instead of a copy and depending on how the array is accessed later, that may be beneficial. If you think it's better to create it this way, I will change it.

Reply:
Let's leave it as is for now.